core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -800,6 +800,30 @@ object TestUtils extends Logging {
     leader
   }
 
+  def waitUntilLeadersAreMade(servers: Seq[KafkaServer], topic: String, partition: Int, timeout: Long = 5000L): Unit = {
+    TestUtils.waitUntilTrue(() => {
+      var res = false
+      servers.foreach(server => {
```

Comment:
Since we set `res` to true if _any_ case is true, we could use `exists` instead of `foldLeft`.